shared/domains/chelonia/chelonia.js

```diff
@@ -451,17 +451,17 @@ async function outEncryptedOrUnencryptedAction (
   contract.metadata.validate(meta, { state, ...gProxy, contractID })
   contract.actions[action].validate(data, { state, ...gProxy, meta, contractID })
   const unencMessage = ({ action, data, meta }: GIOpActionUnencrypted)
-  const message = GIMessage.createV1_0(contractID, previousHEAD,
+  let message = GIMessage.createV1_0(contractID, previousHEAD,
     [
       opType,
       opType === GIMessage.OP_ACTION_UNENCRYPTED ? unencMessage : this.config.encryptFn(unencMessage)
     ],
     manifestHash
     // TODO: add the signature function here to sign the message whether encrypted or not
   )
-  hooks && hooks.prepublish && hooks.prepublish(message)
-  await sbp('chelonia/private/out/publishEvent', message, publishOptions)
-  hooks && hooks.postpublish && hooks.postpublish(message)
+  hooks?.prepublish?.(message)
+  message = await sbp('chelonia/private/out/publishEvent', message, publishOptions)
+  hooks?.postpublish?.(message)
```

Comment:
Nice, I love this simplification of using `?.` and how you added it to many places in the code that needs it!